frontend/src/components/worksheets/NewUpload/NewUpload.jsx

```diff
@@ -41,54 +42,171 @@ class NewUpload extends React.Component<{
         percentComplete: 0,
     }
 
+    inputFolder = React.createRef();
     inputFile = React.createRef();
 
     setFile = () => {
         const files = this.inputFile.current.files;
+
+        if (!files.length) {
+            return;
+        }
+        this.uploadFile(files);
+    }
+
+    setFolder = () => {
+        const files = this.inputFolder.current.files;
         if (!files.length) {
             return;
         }
-        this.uploadFile(files[0]);
+        this.uploadFolder(files);
     }
 
-    uploadFile = (file) => {
-        if (!file) {
+    uploadFile = (files) => {
+        if (!files) {
+            return;
+        }
+        let folderSize = 0;
+        for (const file of files) {
+            folderSize += file.size;
+        }
+        if (folderSize > 2*1024*1024*1024) {
+            alert('File size is large than 2 GB. Please upload your file(s) through CLI.');
             return;
         }
         const { worksheetUUID, after_sort_key } = this.props;
         const { name, description } = this.state;
-        const createBundleData = getDefaultBundleMetadata(name || file.name, description);
+
+        for (const file of files) {
+            const createBundleData = getDefaultBundleMetadata(name || file.name, description);
+            this.setState({

```

Comment:
Right now, you're setting `uploading: true` before each file is uploaded and calling the success/failure callbacks (and `reloadWorksheet`) after each file is uploaded. Can you make it so that the former happens before the for loop (before _all_ files are uploaded) and the latter after _all_ files have been uploaded?